test/jdk/java/lang/System/i18nEnvArg.java

```diff
@@ -0,0 +1,155 @@
+/*
+ * Copyright (c) 2022, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+/*
+ * @test
+ * @bug 8285517
+ * @summary System.getenv() and argument don't return locale dependent data by JEP400
+ * @run main i18nEnvArg
+ */
+
+import java.io.File;
+import java.io.InputStream;
+import java.lang.reflect.Method;
+import java.nio.ByteBuffer;
+import java.nio.charset.Charset;
+import java.util.Arrays;
+import java.util.Map;
+import java.util.StringJoiner;
+
+public class i18nEnvArg {
+    final static String text = "\u6F22\u5B57";
+    final static String javeExe = System.getProperty("java.home")
+        + File.separator
+        + "bin"
+        + File.separator
+        + "java";
+    final static int maxSize = 4096;
+
+    public static void main(String[] args) throws Exception {
```

Comment:
There are 3 main()'s in this test; it would be useful to add a comment indicating the purpose of each.

Reply:
Add comments